indexing-service/src/main/java/org/apache/druid/indexing/overlord/sampler/FirehoseSampler.java

```diff
@@ -256,9 +256,7 @@ public SamplerResponse sample(FirehoseFactory firehoseFactory, DataSchema dataSc
         Map<String, Object> parsed = new HashMap<>();
 
         columnNames.forEach(k -> {
-          if (row.getRaw(k) != null) {
-            parsed.put(k, row.getRaw(k));
-          }
+          parsed.put(k, row.getRaw(k) == null ? "" : row.getRaw(k));
```

Comment:
I think this should just return nulls as is, any null->empty string conversion should have happened already at this point, so any nulls you see here should really be null